### PR TITLE
Add flush duration to status report struct

### PR DESCRIPTION
**Pull request by stephen**
The flush duration is the amount of time it took for the tracer to pack up the records, send them over the network, and rotate buffers. Seeing this measurement is helpful for tuning the min reporting period for the tracer.

Because the reporter loop is serial, this is the minimum time between flushes that the tracer could do. Add a test that this value is measured.

### CODE REVIEW

**Comment by andrewcheelightstep** [comment]
@bg451 Can you review this?

File: events.go [hunk 1]
```diff
@@ -196,6 +200,10 @@ func (s *eventStatusReport) Duration() time.Duration {
 	return s.finishTime.Sub(s.startTime)
 }
 
+func (s *eventStatusReport) FlushDuration() time.Duration {
```
Comment (bg451) [inline events.go]: Could you add a comment that mentions the flush duration is time from flush start to rotating out the buffers, not necessarily just the latency of the network call to flush spans?

Comment (stephen) [reply events.go]: sg, though i think it won't be too helpful to end users unless we add a comment to the interface too - wdyt?

Comment (stephen) [reply events.go]: @bg451, I added godoc comments for each field on the interface, wdyt?

Comment (bg451) [reply events.go]: yep! This is great, thanks!

**Pull request merged**